### PR TITLE
Assorted fixes

### PR DESCRIPTION
**Pull request by dmeltzer**
Fixing some issues as I start clicking through the ui..

1) Fix some n+1 problems on relationship loading.  There's still a few weird ones in the license/license_seat relationship, but I need to stare deeplyer at them before adjusting anything.

2) The dropdown notifications menu was linking to stale/non-existant routes.  Adjust the code to use the named route instead of a direct link.

3) Some bug fixes related to #3891.  I've improved the transformer and the formatter choices to better display checkout targets of non user.  Also remove the Asset::assigneduser() relationship because it was misleading and returning poor results.

One big outstanding issue (my attempt at it is commented out in Asset.php@scopeTextSearch) is handling the search inside of tables to match the location text.  The joins get weird and messy and complicated and I'm having trouble solving those at the moment.

### CODE REVIEW

**Comment by dmeltzer** [comment]
This should also fix #3780 now, I'm not 100% on moving the method verbatim to the api controller, but I think we can look at refactoring later.

**Comment by dmeltzer** [comment]
Some more minor fixes and testing.  Nothing major in this batch, except the commenting out of more assigneduser foo in Asset::scopeByLocationId... Need to determine how to write these things in a query builder friendly way.

File: app/Console/Commands/SendExpectedCheckinAlerts.php [hunk 1]
```diff
@@ -44,14 +44,14 @@ public function fire()
     {
 
         $whenNotify = Carbon::now()->addDays(7);
-        $assets = Asset::with('assigneduser')->whereNotNull('expected_checkin')->where('expected_checkin', '<=', $whenNotify)->get();
+        $assets = Asset::with('assignedTo')->whereNotNull('expected_checkin')->where('expected_checkin', '<=', $whenNotify)->get();
```
Comment (snipe) [inline app/Console/Commands/SendExpectedCheckinAlerts.php]: What's the reason for changing this? assigneduser should have worked (and did in my tests)

File: app/Http/Controllers/Api/ConsumablesController.php [hunk 2]
```diff
@@ -148,4 +148,47 @@ public function destroy($id)
         $consumable->delete();
         return response()->json(Helper::formatStandardApiResponse('success', null,  trans('admin/consumables/message.delete.success')));
     }
+
+        /**
+    * Returns a JSON response containing details on the users associated with this consumable.
```
Comment (snipe) [inline app/Http/Controllers/Api/ConsumablesController.php]: This isn't how we return API data though. We use transformers.

File: app/Http/Controllers/ConsumablesController.php [hunk 3]
```diff
@@ -273,47 +273,4 @@ public function postCheckout($consumableId)
 
     }
 
-
-    /**
-    * Returns a JSON response containing details on the users associated with this consumable.
```
Comment (snipe) [inline app/Http/Controllers/ConsumablesController.php]: I was already working on the consumables API controller - wish you'd have let me know you were poking at it. Really I need any outstanding importer issues handled.

File: app/Console/Commands/SendExpectedCheckinAlerts.php (hunk 1 above)
Comment (dmeltzer) [reply app/Console/Commands/SendExpectedCheckinAlerts.php]: the assigneduser relationship as existed was a lie.  it found the user that matched the assigned_to id, reguardless of the assigned_type.  I tried to modify it to only return something when checked out to a user, but it caused a lot of issues when the relationship returned null.

File: app/Models/SnipeModel.php [hunk 4]
```diff
@@ -82,4 +82,8 @@ public function getDisplayNameAttribute()
     {
         return $this->name;
     }
+
+    public function name() {
```
Comment (snipe) [inline app/Models/SnipeModel.php]: What is this for?

**Pull request merged**